### PR TITLE
fix(integrations): Bring back type="submit" in awsLambdaFunctionSelect

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This regressed in GH-76118

### CODE REVIEW

File: config/tsconfig.base.json [hunk 1]
```diff
@@ -19,7 +19,11 @@
     "moduleResolution": "node",
 
     // We add esnext to lib to pull in types for all newer ECMAScript features
-    "lib": ["esnext", "dom"],
+    "lib": [
+      "esnext",
+      "dom",
+      "dom.iterable"
```
Comment (evanpurkhiser) [inline config/tsconfig.base.json:25]: Added this to allow `.entries` to work on FormData

File: static/app/views/integrationPipeline/awsLambdaFunctionSelect.spec.tsx [hunk 2]
```diff
@@ -16,6 +16,30 @@ describe('AwsLambdaFunctionSelect', () => {
     );
     expect(screen.getByRole('checkbox', {name: 'lambdaB'})).toBeInTheDocument();
     expect(screen.getByRole('button', {name: 'Finish Setup'})).toBeInTheDocument();
-    // TODO: add assertion for form post
+
+    // Uncheck lambda A
+    await userEvent.click(screen.getByRole('checkbox', {name: 'lambdaA'}));
+
+    const {promise, resolve} = Promise.withResolvers<FormData>();
```
Comment (scttcper) [inline static/app/views/integrationPipeline/awsLambdaFunctionSelect.spec.tsx:23]: would avoid until babel can compile it into something older https://caniuse.com/mdn-javascript_builtins_promise_withresolvers

**Review by gggritso — APPROVED** [review]
Unsure if you need to unregister the window event listener at the end of the test, but LGTM otherwise!

**Comment by evanpurkhiser** [comment]
> Unsure if you need to unregister the window event listener at the end of the test, but LGTM otherwise!

Let's do it to be safe!

**Pull request merged**